Core/interface/BareFatJets.hpp

```diff
@@ -12,8 +12,9 @@ class BareFatJets : virtual public BareP4
         ~BareFatJets();
         void clear();
         void init();
-        void defineBranches(TTree *t);
-        virtual void setBranchAddresses(TTree*);
+        // non-prefix versions are defined in BareP4 and will call the following functions with prefix = ""
```

Comment:
I'm afraid you have to redefine also the defineBranches(TTree*t) , or overload the next one with prefix="",
otherwise the BareFatJets class will not be able to call defineBranches(t). I think this is related to how the classes inherithance is written from the compiler.
( so , BareFatJets->defineBranches(t) -> ERROR, while static_cast(BareCollection*)<&b>->defineBranches(t)//ok )